### PR TITLE
#230 fix | used Theme to give dynamic colors

### LINKED ISSUE #230: Bug: Text on login page not visible
## Issue
Some text, e.g., the `Email ID`, `Password` and `New to Resonate` is not clearly visible due to their dark color on the dark background on the application as shown in the screenshot below.

IMG-20240102-WA0009

## Possible Solution
The color of the text can be made lighter so that the user is able to clearly see the text

### PR DESCRIPTION
**Pull request by MrBooring**
## Description

This PR addresses issues with color inconsistency in several key areas of the app, including the Email ID, Password, and "New to Resonate" text in emails, the logout icon in the profile section, and the visibility of the "verify" text in verification emails. The proposed solution involves implementing a unified theme throughout the entire application. Upon approval, I estimate that this task can be completed within 1-2 days.

Fixes #230, Closes #318, #316

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

To verify the changes, I conducted manual testing by following these steps:

Changed the theme settings to both dark mode and light mode.

- Reviewed each affected area, including:

1. Email ID and Password fields in login page.
2. "New to Resonate" text in login page.
3. Logout icon in the profile section.
4. Visibility of the "verify your email" text in verification emails.

Checked if all texts and icons were clearly visible and legible in both dark and light modes.
To reproduce the testing process:

Navigate to the settings or preferences section of the application.
Locate the theme settings and switch between dark mode and light mode.
Access the areas mentioned above, such as the login page for Email ID and Password fields, emails containing the "New to Resonate" text, the profile section for the logout icon, and verification emails for the "verify" text.
Ensure that all texts and icons maintain visibility and legibility in both dark and light modes.
Please let me know if you need further clarification or assistance with testing.

 included screenshots below 

 
Screenshot 2024-03-13 022539

Screenshot 2024-03-13 022654

Screenshot 2024-03-13 022829

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Tag the PR with the appropriate labels

### CODE REVIEW

**Comment by Aarush-Acharya** [comment]
Hey @MrBooring, Good Work 🚀✨

There are some conflicts please do resolve them correctly and push back

I reviewed your code it is good to merge but please do take care of the following as well 

On the verify email_verification_screen there is a 
- animated container having text "verify your email" for providing better interactive user experience
- The bigger container that encapsulates the otp text field and the text around it for enhancing UI 

https://github.com/AOSSIE-Org/Resonate/assets/92685647/d4f54550-d1c9-4cb4-84cb-a951f2f849b5

Rn the colour of the animated container and the text in it along with the parent container wrapping the other widgets is not set so that they are clearly visible in both dark and light themes 

Please play with their colours such that they are visible, look good on both the light and dark modes  hence enhancing the UI

Push after making the suggested changes and resolving the conflicts carefully achieving a clean state, will merge this PR once the ones made prior to this are resolved

**Comment by MrBooring** [comment]
Sure ill look into the mentioned conflicts and try to complete it by tomorrow itself

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by MrBooring** [comment]
@Aarush-Acharya am sorry I was not able to work i got caught up in some other work 
well, I looked at the page you mentioned and I didn't understand why we have that feature, when the user clicks on that headline it disappears.

this is before 

Screenshot_1716063111


and this is after 

Screenshot_1716062955


should I remove the animation where it disappears?

**Comment by Aarush-Acharya** [comment]
@MrBooring no do not remove it, change the background colours of the containers visible when the animation happens it forms an enclosed circle the circle is not visible as the colour is set to white, please adjust the colours of the containers so that they are visible in both dark and white themes

**Comment by MrBooring** [comment]
@Aarush-Acharya 
flutter_01
flutter_02
 
please check this

**Comment by Aarush-Acharya** [comment]
Yes exactly like this good work, 

Just make some more minor changes and will merge it, good going

these are some additional features
- Make sure that the OTP fields change colour according to the app colour, if app colour is purple they should be purple

**Comment by MrBooring** [comment]
Sure ill look into it tonight

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Aarush-Acharya** [comment]
hey @MrBooring thank you for the contribution but this is just one of the suggested changes what about the others, please do complete all of them and push again

**Comment by MrBooring** [comment]
I've made 4 commits

  1.  The original pr
  2.  container having text "verify your email" color issue and dynamic height
  3.  OTP fields
  4.  Merge conflicts

other than this have I missed something

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Aarush-Acharya** [comment]
@MrBooring  yes, please do visit this PR on Github i believe it is because you are interacting via outlook you are not able to see the review changes requested in the form of comments on the code lines 

Please view the PR on github.com for further interactions

**Comment by MrBooring** [comment]
@Aarush-Acharya now I am viewing the PR on github.com I didn't see any comments though yesterday when I was resolving the merge conflict I found one comment to use constant height to make it dynamic and I did that part, if there are other things please help me to find it

**Comment by Aarush-Acharya** [comment]
Screenshot 2024-05-25 at 5 57 40 PM

I meant theese

**Comment by Aarush-Acharya** [comment]
Hey @MrBooring sincere apologies I had not submitted the review it was still in the Draft I belief my bad, please check again

**Comment by Aarush-Acharya** [comment]
Hey @MrBooring sincere apologies I had not submitted the review it was still in the Draft I belief my bad, please check again

**Comment by Aarush-Acharya** [comment]
Hey @MrBooring you there ??

**Comment by MrBooring** [comment]
@Aarush-Acharya sorry am currently out right now i will work on it as soon as i get home sorry for the delay

**Comment by MrBooring** [comment]
@Aarush-Acharya I've completed the review changes requested. if you have any further changes please do let me know

flutter_01
flutter_02

**Comment by Aarush-Acharya** [comment]
Good work, 
@MrBoorin, it looks real good and aligns with what I envisioned,
I believe you missed looking into the Post Script of a comment I made, please implement that as well 
Apart from that everything feels clean and good

File: lib/views/screens/email_verification_screen.dart [hunk 1]
```diff
@@ -229,7 +235,7 @@ class EmailVerificationScreen extends StatelessWidget {
                                         ? Center(
                                             child: Text(
                                               "Verify Your Email",
-                                              style: TextStyle(fontSize: UiSizes.size_14),
+                                              style: TextStyle(fontSize: UiSizes.size_14, color: Colors.black),
```
[the remark this answers is not in the document]
Comment (Aarush-Acharya) [reply lib/views/screens/email_verification_screen.dart]: Hey @MrBooring  please do take care of the Back button hard coded colour as well

**Comment by MrBooring** [comment]
@Aarush-Acharya sorry I missed that one, my bad. I've completed that too

**Comment by Aarush-Acharya** [comment]
Great Ｗork @MrBooring,
I am merging this Pull Request Now

**Comment by MrBooring** [comment]
@Aarush-Acharya happy to help 😊

**Pull request merged**